ddl/index.go

```diff
@@ -99,7 +99,7 @@ func checkPKOnGeneratedColumn(tblInfo *model.TableInfo, indexPartSpecifications
 		}
 		// Virtual columns cannot be used in primary key.
 		if lastCol.IsGenerated() && !lastCol.GeneratedStored {
-			return nil, ErrUnsupportedOnGeneratedColumn.GenWithStackByArgs("Defining a virtual generated column as primary key")
+			return nil, ErrUnsupportedOnGeneratedColumn.GenWithStackByArgs("Primary Key Constraint")
```

Comment:
The error message should be "The primary key cannot be an expression index." and it's error code should be 3756.

Reply:
Will update, thank you

Reply:
![image](https://user-images.githubusercontent.com/17380469/95949452-4dfdf580-0e25-11eb-9b9e-b96f75ccdf46.png)
The err ErrUnsupportedOnGeneratedColumn is wrong here, you should update it to ErrFunctionalIndexPrimaryKey

Reply:
There are two possible situations. First, creating a primary key on a generated column. Two, creating a primary key on an expression index. You should take care of both of them.

Reply:
Hint: if creating a primary key on an expression index, the col.Hidden will be true. You can use this to decide which error to return.